src/main/lrsql/admin/interceptors/account.clj

```diff
@@ -121,6 +144,39 @@
                   :body   {:error (format "An account \"%s\" already exists!"
                                           username)}}))))}))
 
+(def update-admin-password
+  "Set a new password for an admin account."
+  (interceptor
+   {:name ::update-admin-password
+    :enter
+    (fn update-admin-password [ctx]
+      (let [{lrs :com.yetanalytics/lrs
+             {:keys [username old-password new-password]} ::data}
+            ctx
+            {:keys [result]}
+            (adp/-update-admin-password lrs username old-password new-password)]
+        (cond
+          ;; The result is the account ID - success!
+          (uuid? result)
+          (assoc ctx
+                 :response
+                 {:status 200 :body {:account-id result}})
+
+          ;; The given username does not belong to a known account
+          (= :lrsql.admin/missing-account-error result)
+          (assoc (chain/terminate ctx)
+                 :response
+                 {:status 404
+                  :body   {:error (format "The account username \"%s\" does not exist!"
+                                          username)}})
+
+          ;; The old password is not correct.
+          (= :lrsql.admin/invalid-password-error result)
+          (assoc (chain/terminate ctx)
+                 :response
+                 {:status 401
+                  :body   {:error "Invalid Account Credentials"}}))))}))
```

Comment:
Actually one thing @cliffcaseyyet always emphasized when I worked on security implementations is that we should revel as few details as possible when returning security errors. For example, for `authenticate-admin` it would return a 401 error for both error results with a generic error message, and it seems like this should do the same.

Reply:
Though on second look it might be okay to leave them separate here since you'd presumably _already_ be logged in with the username and (old) password. Which means that these should happen very rarely if ever, e.g. if another user concurrently deletes the account, but on those off-chance we'll still need these error cases.

Reply:
To be fair in this case though, they are ALREADY authenticated. I usually advocate that at an authentication border.